### PR TITLE
Use snarkjs API and remove the hardcoding of the snarkjs command line path

### PR DESCRIPTION
**Pull request by yuetloo**
This will fix https://github.com/privacy-scaling-explorations/maci/issues/514

Tested this PR with:
* tests/vanilla/testCheckKeys.sh
* integrationTests/ts/__tests__/suites.test.ts
* clrfund's maci v1 integration [e2e tests](https://github.com/clrfund/monorepo/actions/runs/6660341082)

Note that the snarkjs.d.ts was added because `@types/snarkjs` doesn't work due to a typo causing `path not found` error. If the issue was fixed, the snarkjs.d.ts file can be removed and simply import the `@types/snarkjs` types

I also increased the poll duration in order for the integration tests to pass as they were failing due to the voting period being too short.  This could mean that the changes introduced in this PR made the tests slower because they had to wait for threads to end as opposed to previously just being killed at the end of the snarkjs command line.

### CODE REVIEW

**Review by ctrlc03 — APPROVED** [review]
LGTM, thanks so much for this. 
Gonna first get this PR https://github.com/privacy-scaling-explorations/maci/pull/775 merged - once we have a properly working CI we can then merge this too

**Comment by ctrlc03** [comment]
> This will fix #514
> 
> Tested this PR with:
> 
> * tests/vanilla/testCheckKeys.sh
> * integrationTests/ts/**tests**/suites.test.ts
> * clrfund's maci v1 integration [e2e tests](https://github.com/clrfund/monorepo/actions/runs/6660341082)
> 
> Note that the snarkjs.d.ts was added because `@types/snarkjs` doesn't work due to a typo causing `path not found` error. If the issue was fixed, the snarkjs.d.ts file can be removed and simply import the `@types/snarkjs` types
> 
> I also increased the poll duration in order for the integration tests to pass as they were failing due to the voting period being too short. This could mean that the changes introduced in this PR made the tests slower because they had to wait for threads to end as opposed to previously just being killed at the end of the snarkjs command line.

@yuetloo can you please pull latest dev updates into this branch? that way we can see if all tests run correctly (we fixed some small bugs in the e2e tests)

File: circuits/ts/index.ts [hunk 1]
```diff
@@ -71,69 +83,16 @@ const genProof = (
...
-    //// Generate calldata
-    //const calldataCmd = `node ${snarkjsPath} zkesc ${publicJsonPath} ${proofJsonPath}`
-    //console.log(shelljs.exec(calldataCmd).stdout)
-
-    fs.unlinkSync(proofJsonPath)
-    fs.unlinkSync(publicJsonPath)
-    fs.unlinkSync(vkJsonPath)
-    tmpObj.removeCallback()
-
+const verifyProof = async (publicInputs: any, proof: any, vk: any) => {
```
Comment (baumstern) [inline circuits/ts/index.ts:86]: Looks like this is obvioulsly cleaner. Great work!

**Review by baumstern — APPROVED** [review]
LGTM!

**Comment by yuetloo** [comment]
@ctrlc03 , merged with the latest dev branch and passed integrationTests

**Comment by baumstern** [comment]
There are [a few error in Integration Test](https://github.com/privacy-scaling-explorations/maci/actions/runs/6712182504/job/18243074595?pr=782). 

### Failure one

```
FAIL ts/__tests__/deployPollWithRandomSigner.test.ts (215.786 s)
  Test deployPollWithSigner
    ✕ Full tree, 4 batches, no bribers (210829 ms)

  ● Test deployPollWithSigner › Full tree, 4 batches, no bribers

    expect(received).toBeTruthy()

    Received: false

      31 |     it(test.description, async () => {
      32 |         const result = await executeSuite(test, expect)
    > 33 |         expect(result).toBeTruthy()
         |                        ^
      34 |
      35 |         let caughtException = false
      36 |         try {

      at ts/__tests__/deployPollWithRandomSigner.test.ts:33:24
      at step (ts/__tests__/deployPollWithRandomSigner.test.ts:33:23)
      at Object.next (ts/__tests__/deployPollWithRandomSigner.test.ts:14:53)
      at fulfilled (ts/__tests__/deployPollWithRandomSigner.test.ts:5:58)

  console.log
    stdout: Merging state subroots 1 / 1
    Executed mergeMaciStateAqSubRoots(); gas used: 712418
    Transaction hash: 0xe875bea54d2e3442f78466747772a307ed74ecb1cf3874e82ab73e5034ef7ae8
    
    All state subtrees have been merged.
    Merging subroots to a main state root...
    Executed mergeStateAq(); gas used: 1004922
    Transaction hash: 0x885540f2e8a81cffc1c8efcbab0656a6c7b24fde45d1d0aef2215dd582fada74
    The state tree has been merged.

      at execute (ts/__tests__/suites.ts:28:13)

  console.log
    rm -rf tally.json subsidy.json proofs/

      at execute (ts/__tests__/suites.ts:25:13)

  console.log
    stdout:

      at execute (ts/__tests__/suites.ts:28:13)

  console.log
    node build/index.js genProofs -x 0xb5E58e7CEe446772A8ACD9975D2242dC6B9666E3 -sk macisk.20df7d4e23b1430e28095e4a2856b4db43496beed4365e979b621f5d80382b2 -o 0 -r ~/rapidsnark/build/prover -wp ./zkeys/ProcessMessages_10-2-1-2_test -wt ./zkeys/TallyVotes_10-1-2_test -zp ./zkeys/ProcessMessages_10-2-1-2_test.0.zkey -zt ./zkeys/TallyVotes_10-1-2_test.0.zkey -t tally.json -f proofs/

      at execute (ts/__tests__/suites.ts:25:13)

  console.log
    stdout:

      at execute (ts/__tests__/suites.ts:28:13)

  console.error
    Error: exec failed: Error: ENOENT: no such file or directory, scandir 'proofs/'
        at Object.readdirSync (node:fs:1438:3)
        at /home/runner/work/maci/maci/cli/ts/proveOnChain.ts:210:30
        at step (/home/runner/work/maci/maci/cli/build/proveOnChain.js:33:23)
        at Object.next (/home/runner/work/maci/maci/cli/build/proveOnChain.js:14:53)
        at fulfilled (/home/runner/work/maci/maci/cli/build/proveOnChain.js:5:58)
        at processTicksAndRejections (node:internal/process/task_queues:96:5)
    
        at execute (/home/runner/work/maci/maci/integrationTests/ts/__tests__/suites.ts:30:15)
        at /home/runner/work/maci/maci/integrationTests/ts/__tests__/suites.ts:288:9
        at step (/home/runner/work/maci/maci/integrationTests/ts/__tests__/suites.ts:33:23)
        at Object.next (/home/runner/work/maci/maci/integrationTests/ts/__tests__/suites.ts:14:53)
        at /home/runner/work/maci/maci/integrationTests/ts/__tests__/suites.ts:8:71
        at new Promise (<anonymous>)
        at Object.<anonymous>.__awaiter (/home/runner/work/maci/maci/integrationTests/ts/__tests__/suites.ts:4:12)
        at executeSuite (/home/runner/work/maci/maci/integrationTests/ts/__tests__/suites.ts:39:51)
        at /home/runner/work/maci/maci/integrationTests/ts/__tests__/deployPollWithRandomSigner.test.ts:32:42
        at step (/home/runner/work/maci/maci/integrationTests/ts/__tests__/deployPollWithRandomSigner.test.ts:33:23)

      296 |     }
      297 |     catch(e) {
    > 298 |         console.error(e)
          |                 ^
      299 |         return false
      300 |     } 
      301 |     

      at ts/__tests__/suites.ts:298:17
      at step (ts/__tests__/suites.ts:33:23)
      at Object.next (ts/__tests__/suites.ts:14:53)
      at ts/__tests__/suites.ts:8:71
      at Object.<anonymous>.__awaiter (ts/__tests__/suites.ts:4:12)
      at executeSuite (ts/__tests__/suites.ts:39:51)
```

### Failure two

```
 console.log
    stdout: Merging message subroots 1 / 4
    Executed mergeMaciStateAqSubRoots(); gas used: 861080
    Transaction hash: 0xe66739f7467c3723e19fcfb1b6a09d48559a4faeee35d671f1d0ae28967d3aeb
    
    All message subtrees have been merged.
    Merging subroots to a main message root...
    Executed mergeMessageAq(); gas used: 71043
    Transaction hash: 0x3a260915402b9f717530e1b8a208448d1a56e211332e66daaca1a350f28d27df
    The message tree has been merged.

      at execute (ts/__tests__/suites.ts:28:13)

  console.log
    node build/index.js mergeSignups -x 0x5F600562E388F4d85e52979C499724045DF3Bd61 -o 0

      at execute (ts/__tests__/suites.ts:25:13)

  console.log
    stdout:

      at execute (ts/__tests__/suites.ts:28:13)

  console.error
    Error: exec failed: Error: the transaction failed.
    Error: VM Exception while processing transaction: reverted with reason string 'PollE01'
    
        at execute (/home/runner/work/maci/maci/integrationTests/ts/__tests__/suites.ts:30:15)
        at /home/runner/work/maci/maci/integrationTests/ts/__tests__/suites.ts:198:39
        at step (/home/runner/work/maci/maci/integrationTests/ts/__tests__/suites.ts:33:23)
        at Object.next (/home/runner/work/maci/maci/integrationTests/ts/__tests__/suites.ts:14:53)
        at /home/runner/work/maci/maci/integrationTests/ts/__tests__/suites.ts:8:71
        at new Promise (<anonymous>)
        at Object.<anonymous>.__awaiter (/home/runner/work/maci/maci/integrationTests/ts/__tests__/suites.ts:4:12)
        at executeSuite (/home/runner/work/maci/maci/integrationTests/ts/__tests__/suites.ts:39:51)
        at /home/runner/work/maci/maci/integrationTests/ts/__tests__/suites.test.ts:12:46
        at step (/home/runner/work/maci/maci/integrationTests/ts/__tests__/suites.test.ts:33:23)

      296 |     }
      297 |     catch(e) {
    > 298 |         console.error(e)
          |                 ^
      299 |         return false
      300 |     } 
      301 |     

      at ts/__tests__/suites.ts:298:17
      at step (ts/__tests__/suites.ts:33:23)
      at Object.next (ts/__tests__/suites.ts:14:53)
      at ts/__tests__/suites.ts:8:71
      at Object.<anonymous>.__awaiter (ts/__tests__/suites.ts:4:12)
      at executeSuite (ts/__tests__/suites.ts:39:51)
```

**Pull request merged**